### DISCUSSION TITLE
Reading .eml file

### DISCUSSION THREAD

**Discussion by manstie** [stalwartlabs/mail-parser#79] [Q&A] [answered]
[email-file.zip](https://github.com/stalwartlabs/mail-parser/files/14355684/email-file.zip)

I am attempting to read this .eml file dragged from outlook.

```
    let file = std::fs::read(path)?;
    let message = mail_parser::MessageParser::default()
        .parse(&file)
        .ok_or(format!("Could not parse email file {}", path))?;

    let body = message
        .body_text(0)
        .ok_or(format!("Missing message body in email file {}", path))?;

    log::debug!("Email message body: {}", body);
    ...
```

When reading this email file with code above the entire body logs as:
![image](https://github.com/stalwartlabs/mail-parser/assets/12475567/631d84ee-f64c-451d-9a07-731041da796f)

When opening in outlook, the body appears as:
```
Hey test gratitude cool
```

If I use `.to_string()` on `body` and try to insert it into my database it complains `invalid byte sequence for encoding "UTF8": 0x00`.

The file appears to be in UTF-16LE. Am I doing something wrong?

**Comment by mdecimus** [comment, marked as answer]
The message needs to be in plain text, the EML file you're trying to parse is in binary format.

**Reply by manstie** [reply]
:cry:
